src/ansys/units/quantity.py

```diff
@@ -166,6 +178,39 @@ def __init__(
 
         self._unit = units
 
+        for unit in self._chosen_units:
+            if unit.name != units.name and self.dimensions == unit.dimensions:
+                self._value = self.to(unit).value
+                self._unit = unit
```

Comment:
@ansaboutin in follow up work, I think it would be very nice if we could allow the call to `preferred units` to specify association with a specific quantity map. Then a user could say associate energy -> joule and torque -> Nm. I realise that there's not much time left for you to look into that.

Reply:
Would this be handled easier with a custom map? I am unsure how this could be specified:
```
meter = Unit("m")
N = Unit("N")
unit = N * meter
```
Should Unit stay as Nm or change to J?

Reply:
I think I agree with Sean.  You might want to pass a quantity name from the map table instead of a unit and use the units of the quantity name in the map table.

Reply:
Staring at this a little more I wonder if we might want to always use quantity names.  The thing is that if we match based on base units this code has to hunt through the list of preferred units for a matching unit with the same base dimensions.  Might be a little slow no?

Reply:
In future work would it make sense for `Unit(units="Torque")` to make a unit from the map?

Reply:
To be discussed further and handled in future work.